### PR TITLE
[Program Plugin] Fix UWP icon missing issue

### PR DESCRIPTION
**Pull request by VictoriousRaptor**
Program plugin can't get icons for some uwp programs because of complex icon filenames or path finding issue. Tried to fix it.

### CODE REVIEW

File: Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs [hunk 1]
```diff
@@ -601,75 +601,94 @@ internal string LogoPathFromUri(string uri)
...
                 // windows 8 https://msdn.microsoft.com/en-us/library/windows/apps/br211475.aspx
 
-                string path;
-                if (uri.Contains("\\"))
-                {
-                    path = Path.Combine(Package.Location, uri);
-                }
-                else
-                {
-                    // for C:\Windows\MiracastView etc
```
Comment (VictoriousRaptor) [inline Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs:611]: Not sure why we needed it. C:\Windows\MiracastView shouldn't be a uwp program right?

File: Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs [hunk 2]
```diff
@@ -601,75 +601,94 @@ internal string LogoPathFromUri(string uri)
...
+                    var files = Directory.EnumerateFiles(logoDir);
+
+                    // Just ignore all qulifiers
+                    // select like logo[xxx].png
+                    // https://learn.microsoft.com/en-us/windows/uwp/app-resources/tailor-resources-lang-scale-contrast
+                    var selected = files.FirstOrDefault(file => 
+                        Path.GetFileName(file).StartsWith(logoNamePrefix) && extension == Path.GetExtension(file)
+                    );
+
+                    //var scaleFactors = new Dictionary<PackageVersion, List<int>>
```
Comment (taooceros) [inline Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs:644]: why do we remove them?

File: Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs (hunk 1 above)
Comment (taooceros) [reply Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs:611]: Maybe it is in some older version of win10

File: Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs (hunk 2 above)
Comment (VictoriousRaptor) [reply Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs:644]: I can't find any guarantee in the doc that scale factors comes last to the qualifiers (see [here](https://learn.microsoft.com/en-us/windows/uwp/app-resources/images-tailored-for-scale-theme-contrast)). And we don't acutally need scale factor in our plugin.

Comment (taooceros) [reply Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs:644]: true, let's remove them. We shall fetch all asset information in the manifest file.

File: Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs (hunk 1 above)
Comment (VictoriousRaptor) [reply Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs:611]: I can't even find it on my pc.

Comment (taooceros) [reply Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs:611]: I neither

Comment (taooceros) [reply Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs:611]: Check whether file exists? That shall be good.

Comment (VictoriousRaptor) [reply Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs:611]: Check the new commit please.

File: Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs [hunk 3]
```diff
@@ -601,92 +601,114 @@ internal string LogoPathFromUri(string uri)
...
-                    if (!string.IsNullOrEmpty(selected))
-                    {
-                        return selected;
+                        var files = Directory.EnumerateFiles(logoDir);
+
+                        // Currently we don't care which one to choose
+                        // Just ignore all qualifiers
+                        // select like logo.[xxx_yyy].png
+                        // https://learn.microsoft.com/en-us/windows/uwp/app-resources/tailor-resources-lang-scale-contrast
+                        var selected = files.FirstOrDefault(file =>
```
Comment (taooceros) [inline Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs:642]: maybe we should prefer $44\times44$ if possible to reduce image size? Though it's not a big deal if it is hard.

File: Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs [hunk 4]
```diff
@@ -601,92 +601,114 @@ internal string LogoPathFromUri(string uri)
...
+                            Path.GetFileName(file).StartsWith(logoNamePrefix) && extension == Path.GetExtension(file)
+                        );
+
+                        //var scaleFactors = new Dictionary<PackageVersion, List<int>>
```
Comment (taooceros) [inline Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs:646]: Please remove them if not needed.

**Review by taooceros — COMMENTED** [review]
Check comment and we can merge after that. Could you share an example that the logo won't be loaded so I can give it a test?

**Comment by VictoriousRaptor** [comment]
> Check comment and we can merge after that. Could you share an example that the logo won't be loaded so I can give it a test?

windows security center

File: Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs (hunk 3 above)
Comment (VictoriousRaptor) [reply Plugins/Flow.Launcher.Plugin.Program/Programs/UWP.cs]: Try to find closest to 44x44 in the new commit. It seems to work well.

**Comment by taooceros** [comment]
@VictoriousRaptor What do you think about just use the icoPath property instead of the callback to let ImageLoader loads the image? I think it's no difference than a specific logo (we can adjust the imageloader a bit to load the actual image but currently icon also works I think).

**Comment by VictoriousRaptor** [comment]
> @VictoriousRaptor What do you think about just use the icoPath property instead of the callback to let ImageLoader loads the image? I think it's no difference than a specific logo (we can adjust the imageloader a bit to load the actual image but currently icon also works I think).

Don't quite know how the current callback works. What is the purpose of this change?

**Comment by taooceros** [comment]
> > @VictoriousRaptor What do you think about just use the icoPath property instead of the callback to let ImageLoader loads the image? I think it's no difference than a specific logo (we can adjust the imageloader a bit to load the actual image but currently icon also works I think).
> 
> Don't quite know how the current callback works. What is the purpose of this change?

As you have commented. Let the imageloader loads the icon?

The current callback handles that separately, but I think it is unnecessary since there is a logo image there.

**Comment by VictoriousRaptor** [comment]
> > > @VictoriousRaptor What do you think about just use the icoPath property instead of the callback to let ImageLoader loads the image? I think it's no difference than a specific logo (we can adjust the imageloader a bit to load the actual image but currently icon also works I think).
> > 
> > 
> > Don't quite know how the current callback works. What is the purpose of this change?
> 
> As you have commented. Let the imageloader loads the icon?
> 
> The current callback handles that separately, but I think it is unnecessary since there is a logo image there.

I think it's fine. But current callback make the image plated so we need to beware of that.

**Comment by taooceros** [comment]
yeah I am ok with the current change. Let's leave it to the future~

**Review by taooceros — APPROVED** [review]
LGTM! Sorry for the late testing....

**Pull request merged**